### PR TITLE
Envoy Element Code Generation

### PR DESCRIPTION
**Pull request by livingshade**
Generate envoy `lib.rs` from adn specs.
Currently only support ACL, need to use `aclping.adn` as frontend. 
Then it can be directly compiled in output_dir.

### CODE REVIEW

File: compiler/element/backend/envoy/boilerplate.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+use proxy_wasm::traits::{{Context, HttpContext}};
+use proxy_wasm::types::{{Action, LogLevel}};
+use proxy_wasm::traits::RootContext;
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+use std::sync::Mutex;
+use prost::Message;
+
+pub mod ping {{
+    include!(concat!(env!("OUT_DIR"), "/ping_pb.rs"));
```
Comment (Romero027) [inline compiler/element/backend/envoy/boilerplate.py:11]: I feel like `"./ping_pb.rs"` should not be hard-code?

**Review by Romero027 — COMMENTED** [review]
Looks great! Thanks a lot! 

I think we should have "./ping_pb.pb" as configuration.

File: compiler/element/backend/envoy/boilerplate.py (hunk 1 above)
Comment (livingshade) [reply compiler/element/backend/envoy/boilerplate.py:11]: I agree. How about we set up a issue for this?

Comment (Romero027) [reply compiler/element/backend/envoy/boilerplate.py:11]: Sure, can you set it up?

**Comment by Romero027** [comment]
This is already a pretty big PR. Feel free to merge this first! @livingshade

File: compiler/element/README.md [hunk 2]
```diff
@@ -7,11 +7,14 @@ Element compiler convert ADN program to an IR. From IR, we can infer the element
 ```
 cd compiler
 export PYTHONPATH=$PYTHONPATH:$(pwd):$(dirname $(pwd))
-python ir_main.py -e acl -v True
-# v for verbose, which is `false` by default
-# refer to example/match-action for more engine names
+python ir_main.py -e acl -v True -b envoy -p c
```
Comment (Romero027) [inline compiler/element/README.md:10]: can you rename `ir_main.py` to `element_main.py`?

**Review by Romero027 — APPROVED** [review]
Just a small nit

**Pull request merged**